cwltool/pathmapper.py

```diff
@@ -77,6 +77,19 @@ def addLocation(d):
         if "basename" not in d:
             d["basename"] = os.path.basename(urllib.request.url2pathname(path))
 
+        if d["class"] == "File":
+            if "nameroot" or "nameext" not in d:
```

Comment:
You can also remove the conditional.  As mentioned earlier we can set `nameroot` and `nameext` unconditionally.

FYI that construction doesn't do what you want:

```
if "x" or "y" in "abc":
  print "yes"
```

This will always print "yes" because the expression is actually `bool("x") or bool("y" in "abc")` and non-empty strings are `True`.